app/services/user_treatment.rb

```diff
@@ -0,0 +1,19 @@
+class UserTreatment
+
+  def call(user: user, type: type)
+    @type = type
+    relation = Hash.new {|h,k| h[k] = @type }
+    relation[@type.class.to_s.underscore]
```

Comment:
```@type``` appears to always be a TrimType... so this is always relation["trim_type"], and then your hash is...
relation["trim_type"] = trim_type...